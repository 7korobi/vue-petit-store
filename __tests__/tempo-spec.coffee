{ to_timer, to_msec, to_sec, to_relative_time_distance } = require "../lib/index.min"

describe "to_relative_time_distance", =>
  test 'snapshot edge in 52bit', =>
    expect to_relative_time_distance -0xfffffffffffff
    .toMatchSnapshot()
    expect to_relative_time_distance  0xfffffffffffff
    .toMatchSnapshot()
    return
  test 'snapshot edge out 52bit', =>
    expect to_relative_time_distance -0x10000000000000
    .toMatchSnapshot()
    expect to_relative_time_distance  0x10000000000000
    .toMatchSnapshot()
    return

  test 'snapshot distance', =>
    expect to_relative_time_distance -to_msec "1秒"
    .toMatchSnapshot()
    expect to_relative_time_distance -to_msec "59分59秒"
    .toMatchSnapshot()
    expect to_relative_time_distance -to_msec "23時間59分59秒"
    .toMatchSnapshot()
    expect to_relative_time_distance -to_msec "6日23時間59分59秒"
    .toMatchSnapshot()
    expect to_relative_time_distance -to_msec "4週1日23時間59分59秒"
    .toMatchSnapshot()
    expect to_relative_time_distance -to_msec "52週1日5時48分45秒"
    .toMatchSnapshot()

    expect to_relative_time_distance to_msec "1秒"
    .toMatchSnapshot()
    expect to_relative_time_distance to_msec "1分"
    .toMatchSnapshot()
    expect to_relative_time_distance to_msec "1h"
    .toMatchSnapshot()
    expect to_relative_time_distance to_msec "1日"
    .toMatchSnapshot()
    expect to_relative_time_distance to_msec "1週間"
    .toMatchSnapshot()
    expect to_relative_time_distance to_msec "1年"
    .toMatchSnapshot()
    return

  test 'snapshot 半', =>
    expect to_timer to_msec "1秒半"
    .toMatchSnapshot()
    expect to_timer to_msec "1分半"
    .toMatchSnapshot()
    expect to_timer to_msec "1時間半"
    .toMatchSnapshot()
    expect to_timer to_msec "1日半"
    .toMatchSnapshot()
    expect to_timer to_msec "1年半"
    .toMatchSnapshot()
    return
  return

describe "to_timer and to_msec", =>
  test 'value eng', =>
    expect to_timer to_msec "1秒"
    .toMatchSnapshot()
    expect to_timer to_msec "1分1秒"
    .toMatchSnapshot()
    expect to_timer to_msec "1時1分1秒"
    .toMatchSnapshot()
    expect to_timer to_msec "1日1時1分1秒"
    .toMatchSnapshot()
    expect to_timer to_msec "1週1日1時1分1秒"
    .toMatchSnapshot()
    expect to_timer to_msec "1年1週1日1時1分1秒"
    .toMatchSnapshot()
    return
  test 'value jpn', =>
    expect to_timer to_msec("1秒"), 0
    .toMatchSnapshot()
    expect to_timer to_msec("1分1秒"), 0
    .toMatchSnapshot()
    expect to_timer to_msec("1時1分1秒"), 0
    .toMatchSnapshot()
    expect to_timer to_msec("1日1時1分1秒"), 0
    .toMatchSnapshot()
    expect to_timer to_msec("1週1日1時1分1秒"), 0
    .toMatchSnapshot()
    expect to_timer to_msec("1年1週1日1時1分1秒"), 0
    .toMatchSnapshot()
    expect to_timer to_msec("1年") - 1, 0
    .toMatchSnapshot()
    return
  return