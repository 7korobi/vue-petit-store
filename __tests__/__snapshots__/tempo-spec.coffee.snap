// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tempo daily 1`] = `
Tempo {
  "args": Array [
    "1d",
  ],
  "callback": [Function],
  "tempo": Object {
    "last_at": -32400000,
    "next_at": 54000000,
    "now_idx": 4,
    "remain": 54000000,
    "since": 32400000,
    "size": 86400000,
    "timeout": 54000000,
    "write_at": 0,
    "zero": -378000000,
  },
}
`;

exports[`Tempo daily 2`] = `
Tempo {
  "args": Array [
    "1d",
  ],
  "callback": [Function],
  "tempo": Object {
    "last_at": 1575126000000,
    "next_at": 1575212400000,
    "now_idx": 18235,
    "remain": 32478726,
    "since": 53921274,
    "size": 86400000,
    "timeout": 32478726,
    "write_at": 1575179921274,
    "zero": -378000000,
  },
}
`;

exports[`Tempo daily same time 1`] = `
Tempo {
  "args": Array [
    "1d",
    "15時40分",
  ],
  "callback": [Function],
  "tempo": Object {
    "last_at": -62400000,
    "next_at": 24000000,
    "now_idx": 3,
    "remain": 24000000,
    "since": 62400000,
    "size": 86400000,
    "timeout": 24000000,
    "write_at": 0,
    "zero": -321600000,
  },
}
`;

exports[`Tempo daily same time 2`] = `
Tempo {
  "args": Array [
    "1d",
    "15時40分",
  ],
  "callback": [Function],
  "tempo": Object {
    "last_at": 1575096000000,
    "next_at": 1575182400000,
    "now_idx": 18234,
    "remain": 2478725,
    "since": 83921275,
    "size": 86400000,
    "timeout": 2478725,
    "write_at": 1575179921275,
    "zero": -321600000,
  },
}
`;

exports[`Tempo hourly 1`] = `
Tempo {
  "args": Array [
    "1h",
  ],
  "callback": [Function],
  "tempo": Object {
    "last_at": 0,
    "next_at": 3600000,
    "now_idx": 105,
    "remain": 3600000,
    "since": 0,
    "size": 3600000,
    "timeout": 3600000,
    "write_at": 0,
    "zero": -378000000,
  },
}
`;

exports[`Tempo hourly 2`] = `
Tempo {
  "args": Array [
    "1h",
  ],
  "callback": [Function],
  "tempo": Object {
    "last_at": 1575176400000,
    "next_at": 1575180000000,
    "now_idx": 437654,
    "remain": 78728,
    "since": 3521272,
    "size": 3600000,
    "timeout": 78728,
    "write_at": 1575179921272,
    "zero": -378000000,
  },
}
`;

exports[`Tempo weekly same time 1`] = `
Tempo {
  "args": Array [
    "1w",
    "1d",
  ],
  "callback": [Function],
  "tempo": Object {
    "last_at": -291600000,
    "next_at": 313200000,
    "now_idx": 0,
    "remain": 313200000,
    "since": 291600000,
    "size": 604800000,
    "timeout": 313200000,
    "write_at": 0,
    "zero": -291600000,
  },
}
`;

exports[`Tempo weekly same time 2`] = `
Tempo {
  "args": Array [
    "1w",
    "1d",
  ],
  "callback": [Function],
  "tempo": Object {
    "last_at": 1574607600000,
    "next_at": 1575212400000,
    "now_idx": 2604,
    "remain": 32478725,
    "since": 572321275,
    "size": 604800000,
    "timeout": 32478725,
    "write_at": 1575179921275,
    "zero": -291600000,
  },
}
`;

exports[`to_relative_time_distance snapshot distance 1`] = `
Array [
  25000,
  25000,
  25000,
  "今",
]
`;

exports[`to_relative_time_distance snapshot distance 2`] = `
Array [
  -60000,
  60000,
  60000,
  "%s分後",
]
`;

exports[`to_relative_time_distance snapshot distance 3`] = `
Array [
  -3600000,
  3600000,
  3600000,
  "%s時間後",
]
`;

exports[`to_relative_time_distance snapshot distance 4`] = `
Array [
  -86400000,
  86400000,
  86400000,
  "%s日後",
]
`;

exports[`to_relative_time_distance snapshot distance 5`] = `
Array [
  -604800000,
  604800000,
  604800000,
  "%s週間後",
]
`;

exports[`to_relative_time_distance snapshot distance 6`] = `
Array [
  -2592000000,
  2147483647,
  2592000000,
  "%sヶ月後",
]
`;

exports[`to_relative_time_distance snapshot distance 7`] = `
Array [
  25000,
  25000,
  25000,
  "今",
]
`;

exports[`to_relative_time_distance snapshot distance 8`] = `
Array [
  3600000,
  60000,
  60000,
  "%s分前",
]
`;

exports[`to_relative_time_distance snapshot distance 9`] = `
Array [
  86400000,
  3600000,
  3600000,
  "%s時間前",
]
`;

exports[`to_relative_time_distance snapshot distance 10`] = `
Array [
  604800000,
  86400000,
  86400000,
  "%s日前",
]
`;

exports[`to_relative_time_distance snapshot distance 11`] = `
Array [
  2592000000,
  604800000,
  604800000,
  "%s週間前",
]
`;

exports[`to_relative_time_distance snapshot distance 12`] = `
Array [
  4503599627370495,
  2147483647,
  31556925147,
  "%s年前",
]
`;

exports[`to_relative_time_distance snapshot edge in 52bit 1`] = `
Array [
  -31556925147,
  2147483647,
  31556925147,
  "%s年後",
]
`;

exports[`to_relative_time_distance snapshot edge in 52bit 2`] = `
Array [
  Infinity,
  2147483647,
  4503599627370495,
  "昔",
]
`;

exports[`to_relative_time_distance snapshot edge out 52bit 1`] = `
Array [
  -4503599627370495,
  2147483647,
  31556925147,
  "？？？",
]
`;

exports[`to_relative_time_distance snapshot edge out 52bit 2`] = `
Array [
  -4503599627370495,
  2147483647,
  31556925147,
  "？？？",
]
`;

exports[`to_relative_time_distance snapshot 半 1`] = `"1s"`;

exports[`to_relative_time_distance snapshot 半 2`] = `"1m30s"`;

exports[`to_relative_time_distance snapshot 半 3`] = `"1h30m"`;

exports[`to_relative_time_distance snapshot 半 4`] = `"1d12h"`;

exports[`to_relative_time_distance snapshot 半 5`] = `"1y26w14h54m22s"`;

exports[`to_timer and to_msec value eng 1`] = `"1s"`;

exports[`to_timer and to_msec value eng 2`] = `"1m1s"`;

exports[`to_timer and to_msec value eng 3`] = `"1h1m1s"`;

exports[`to_timer and to_msec value eng 4`] = `"1d1h1m1s"`;

exports[`to_timer and to_msec value eng 5`] = `"1w1d1h1m1s"`;

exports[`to_timer and to_msec value eng 6`] = `"1y1w1d1h1m1s"`;

exports[`to_timer and to_msec value jpn 1`] = `"1秒"`;

exports[`to_timer and to_msec value jpn 2`] = `"1分1秒"`;

exports[`to_timer and to_msec value jpn 3`] = `"1時1分1秒"`;

exports[`to_timer and to_msec value jpn 4`] = `"1日1時1分1秒"`;

exports[`to_timer and to_msec value jpn 5`] = `"1週1日1時1分1秒"`;

exports[`to_timer and to_msec value jpn 6`] = `"1年1週1日1時1分1秒"`;

exports[`to_timer and to_msec value jpn 7`] = `"52週1日5時48分45秒"`;
